### PR TITLE
Add new FAQ answer and update misc copy

### PR DESCRIPTION
**Pull request by nonparibus**
This PR contains a number of changes to BP site copy:

1. It updates the hiring banner copy to remove reference to WordPress (with plugin development now underway) and to stress the block-building nature of the open Block Protocol roles that [HASH](https://hash.dev/) is funding, as this is likely to be of interest to prospective applicants -- and appealing.
1. It changes various references to the 'registry' to the 'Hub' and corrects errant capitalization of terms in the copy that shouldn't have been proper-cased.
1. It adds a new FAQ answer that in part seeks to explain the major strategies through which the decentralized type system is made workable. A more full answer will be written up as part of a forthcoming blog post (tentatively titled 'encouraging type convergence') to be published on the [HASH Developer Blog](https://hash.dev/blog)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202765662321721
  - https://app.asana.com/0/0/1202336505543679

### CODE REVIEW

File: site/src/_pages/docs/0_index.mdx [hunk 1]
```diff
@@ -6,7 +6,7 @@ The Block Protocol is an open standard for building block-based interfaces. It e
 
 The following pages provide a practical introduction for developers who want to:
 
-- [develop blocks](/docs/developing-blocks) and publish them to the block registry
+- [develop blocks](/docs/developing-blocks) and publish them to the Hub
```
Comment (Alfred-Mountfield) [inline site/src/_pages/docs/0_index.mdx:9]: Do you perhaps want to link to https://blockprotocol.org/hub?

File: site/src/_pages/docs/4_faq.mdx [hunk 2]
```diff
@@ -244,9 +244,19 @@ All schemas created on the Block Protocol website are currently public. In futur
...
+1.  Not everything is included in the [Schema.org](https://schema.org) ontology. For example, although _animal shelters_ and _pet stores_ exist, _animals_ and _pets_ themselves are missing entirely. This arguably reflects the priorities of schema.org’s maintainers (notably corporates, and search engines at that).
+
+</FAQ>
+
+<FAQ question="Doesn't a decentralized schema registry result in problems?">
+
+Moving beyond a single set of constrained schemas that rarely change introduces new issues. We believe that all of these issues can be mitigated and that the benefits of an open ecosystem of well-described, discoverable types grealy outweigh the drawbacks.
```
Comment (Alfred-Mountfield) [inline site/src/_pages/docs/4_faq.mdx]:
```suggestion
Moving beyond a single set of constrained schemas that rarely change introduces new issues. We believe that all of these issues can be mitigated and that the benefits of an open ecosystem of well-described, discoverable types greatly outweigh the drawbacks.
```

File: site/src/_pages/docs/4_faq.mdx [hunk 3]
```diff
@@ -244,9 +244,19 @@ All schemas created on the Block Protocol website are currently public. In futur
...
+
+  1. **Reusability of types:** the type system supports a high degree of composability, with data types and property types indvidually defineable and addressable from within different entity types and contexts.
```
Comment (Alfred-Mountfield) [inline site/src/_pages/docs/4_faq.mdx:257]:
```suggestion
  1. **Reusability of types:** the type system supports a high degree of composability, with data types and property types individually definable and addressable from within different entity types and contexts.
```

Although I'm not sure definable is a word but https://www.merriam-webster.com/dictionary/definable

File: site/src/_pages/docs/4_faq.mdx [hunk 4]
```diff
@@ -244,9 +244,19 @@ All schemas created on the Block Protocol website are currently public. In futur
...
+  1. **Convergence on types:** the type editors developed by [HASH](https://hash.dev/) as part of their ontology manager include functionality to encourage users to where utilize and extend or otherwise crosswalk to existing types. These best-practice editors are open-source and are free to use within any other embedding application. We will be replacing the Block Protocol schema editor with these new type editors shortly.
```
Comment (Alfred-Mountfield) [inline site/src/_pages/docs/4_faq.mdx:258]: Unsure of what "to where utilize" means

File: site/src/components/footer.tsx [hunk 5]
```diff
@@ -210,7 +210,7 @@ export const Footer: FunctionComponent<FooterProps> = ({
                     color: ({ palette }) => palette.gray[50],
                   }}
                 >
-                  The open-source specification and registry
+                  The open-source specification and hub
```
Comment (Alfred-Mountfield) [inline site/src/components/footer.tsx:213]: "Hub" is capitalised elsewhere, is this intended

**Review by Alfred-Mountfield — CHANGES_REQUESTED** [review]

File: site/src/_pages/docs/4_faq.mdx [hunk 6]
```diff
@@ -252,7 +252,7 @@ All schemas created on the Block Protocol website are currently public. In futur
 
 <FAQ question="Doesn't a decentralized schema registry result in problems?">
 
-Moving beyond a single set of constrained schemas that rarely change introduces new issues. We believe that all of these issues can be mitigated and that the benefits of an open ecosystem of well-described, discoverable types grealy outweigh the drawbacks.
+Moving beyond a single set of constrained schemas that rarely change introduces a number of issues. However, we believe that all of these can be mitigated and that the benefits of an open ecosystem of well-described, discoverable, and end-user defineable types greatly outweighs any potential drawbacks.
```
Comment (Alfred-Mountfield) [inline site/src/_pages/docs/4_faq.mdx:255]: Same comment about 'defineable'

Also I think "outweigh" was correct as you're referring to "benefit_s_"? I'm unsure

i.e.
"the benefits outweigh"
as opposed to
"the benefits outweighs"

File: site/src/components/footer.tsx (hunk 5 above)
Comment (nonparibus) [reply site/src/components/footer.tsx:213]: Yes, see commit description

File: site/src/_pages/docs/4_faq.mdx (hunk 6 above)
Comment (nonparibus) [reply site/src/_pages/docs/4_faq.mdx]: Addressed in https://github.com/blockprotocol/blockprotocol/commit/fa28251b8df96c64802a0ea8208d98d1a4abc056 - ty for spotting the typo introduced in the previous typo fix commit 🙃

Comment (nonparibus) [reply site/src/_pages/docs/4_faq.mdx]: `definable` -> `definable` fixed in https://github.com/blockprotocol/blockprotocol/commit/90953b8cbf5266d7599abbc08fbf87aec679d724

File: site/src/_pages/docs/4_faq.mdx (hunk 3 above)
Comment (nonparibus) [reply site/src/_pages/docs/4_faq.mdx]: `definable` -> `definable` fixed in https://github.com/blockprotocol/blockprotocol/commit/90953b8cbf5266d7599abbc08fbf87aec679d724

File: site/src/_pages/docs/4_faq.mdx (hunk 2 above)
Comment (nonparibus) [reply site/src/_pages/docs/4_faq.mdx]: Addressed in https://github.com/blockprotocol/blockprotocol/pull/624/commits/3e19942e1b1d5685cd9d6861b0d9907e0ce1c854

File: site/src/_pages/docs/4_faq.mdx (hunk 4 above)
Comment (nonparibus) [reply site/src/_pages/docs/4_faq.mdx]: Addressed in https://github.com/blockprotocol/blockprotocol/commit/84f1ec523ecbe7631b4570f3b564ebf5054991be -- thanks for spotting. Does it make sense now?

File: site/src/_pages/docs/0_index.mdx (hunk 1 above)
Comment (nonparibus) [reply site/src/_pages/docs/0_index.mdx]: Done in https://github.com/blockprotocol/blockprotocol/commit/c2dd3314d99913ea5334b2d0e905cefa8df14c42

File: site/src/_pages/docs/4_faq.mdx (hunk 4 above)
Comment (Alfred-Mountfield) [reply site/src/_pages/docs/4_faq.mdx]: Sure, thanks :)

**Review by Alfred-Mountfield — COMMENTED** [review]
Prettier needs to be reran, and branch updated. But otherwise it looks good to me

**Review by nathggns — APPROVED** [review]

**Pull request merged**